mongo/mongo.go

```diff
@@ -542,9 +526,26 @@ func ensureServer(args EnsureServerParams, mongoKernelTweaks map[string]string)
 		return fmt.Errorf("error creating oplog files: %v", err)
 	}
 
-	if err := service.InstallAndStart(svc); err != nil {
+	upSvc, ok := svc.(service.UpgradableService)
```

Comment:
I think the only service that implements UpgradableService is systemd.Service, isn't it? That means this code won't work on trusty (where upstart's used).

After discussing with you, I think this is the wrong place to fix this problem. It sounds like what was happening was: 
* you've changed systemd.Service so that it looks in /lib/systemd/system instead of /var/lib/juju/init for the service files.
* on an upgrade, svc.Installed would return true because the mongod service was listed by systemctl
* but then on line 518 the Exists call would return an error - it would fail to read the service file because it was looking in /var/lib/juju/init rather than /lib/systemd/system.

If we make systemd.Service able to handle this error and return false in that specific case, then the InstallAndStart call that happens below will do what we want. Can you do that instead please?